src/one_off.rs

```diff
@@ -31,7 +34,7 @@ use super::Request;
 /// let string = surf::get("https://httpbin.org/get").recv_string().await?;
 /// # Ok(()) }
 /// ```
-pub fn get(uri: impl AsRef<str>) -> Request<NativeClient> {
+pub fn get(uri: impl AsRef<str>) -> Request<Client> {
```

Comment:
Because we already have a `Client` struct, introducing another similarly named client in the same range of APIs may be confusing. I don't see any _great_ options here, but perhaps even something like this may work:

```rust
#[cfg(feature = "hyper-client")]
use super::http_client::hyper::HyperClient as NativeClient;
```

Or alternatively we could use a different name (if there are any better suggestions).

Reply:
How about `HttpClient`?

Reply:
That's the name of the trait we're planning to expose that's used by `HyperClient` etc. It'll be public through the `with_client` method.. Uhhh, maybe it's okay to leave the one-off methods for now and only expose it through the `Client` instance?

Reply:
It would be a shame if the one-off methods don't work, especially because the defaults (= native-client) fail to compile on macOS, but it all does work with hyper.

I'm fine with having `HyperClient as NativeClient`.
The docs will correctly expand this anyway (and docs for all things at the same time doesn't seem easily possible anyway)